config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -34,11 +37,152 @@
+                        LOGGER.log(Level.FINE, "Annotated entity list : " + entityList);
+                        annotatedElements.put(annotatedEntity.getBundleName(), bundleDocumentBuilder.build(document, entityList));
+                    }
+                });
+            }
+            return annotatedElements;
+        }
+
+        Map<String, Element> bundles = new HashMap<>();
+        bundles.put(bundleName + '-' + bundleVersion, bundleDocumentBuilder.build(document, entities));
+        return bundles;
+    }
+
+    private List<Entity> getEntityDependencies(String annotatedEntityName, String annotatedEntityType, String policyNameWithPath, List<Entity> entities, Bundle bundle) {
+        List<Entity> entityDependenciesList = new ArrayList<>();
+        Map<Dependency, List<Dependency>> dependencyListMap = bundle.getDependencyMap();
+        if (dependencyListMap != null) {
+            int pathIndex = policyNameWithPath.lastIndexOf("/");
+            final String policyName = pathIndex > -1 ? policyNameWithPath.substring(pathIndex + 1) : policyNameWithPath;
+            Set<Map.Entry<Dependency, List<Dependency>>> entrySet = dependencyListMap.entrySet();
+            for (Map.Entry<Dependency, List<Dependency>> entry : entrySet) {
+                final Dependency dependencyParent = entry.getKey();
+                if (dependencyParent.getName().equals(policyName)) {
+                    //Add the dependant folders first
+                    final Map<String, Policy> entityMap = bundle.getPolicies();
+                    if (entityMap != null) {
+                        final GatewayEntity policyEntity = entityMap.get(policyNameWithPath);
+                        populatedDependentFolders(entityDependenciesList, entities, policyEntity);
+                    }
+
+                    //Add the policy dependencies
+                    for (Dependency dependency : entry.getValue()) {
+                        if (!dependency.getName().equals(annotatedEntityName) && !dependency.getType().equals(annotatedEntityType)) {
+                            for (Entity entity : entities) {
+                                int index = entity.getName().lastIndexOf("/");
+                                final String entityName = index > -1 ? entity.getName().substring(index + 1) : entity.getName();
+                                if (dependency.getName().equals(entityName) && dependency.getType().equals(entity.getType())) {
+                                    entityDependenciesList.add(entity);
+                                }
+                            }
+                        }
+                    }
+
+                    //Add the parent entities
+                    final List<Entity> parentEntities = entities.stream().filter(entity -> policyNameWithPath.equals(entity.getName()) || annotatedEntityName.equals(entity.getName())).collect(Collectors.toList());
```

Comment:
annotatedEntityName.equals(entity.getName()))

In the above condition entity.getName() returns namewithpath where as annotatedEntityName does not contain path
this might fail policy and service

Reply:
For annotated encass this is not an issue, but will add the check